### DISCUSSION TITLE
Gif -> Gif conversion breaks when specifying a start frame

### DISCUSSION THREAD

**Discussion by 3ternal** [ImageMagick/ImageMagick#4802] [Help] [answered]
I'm trying to set up a workflow for optimizing gifs without using a cloud service. Note that **I'm on Windows 10**. I updated to version 7.1.0-22 and confirmed this with the -version command.

This is the command I'm testing with:
`magick convert input.gif[5-25] output.gif`

If I do this, it completely breaks the colors, and a large part of the background becomes transparent: https://i.imgur.com/bPLrPYV.gif

Interestingly, the command works correctly if I do `magick convert input.gif[0-25] output.gif`
See example: https://i.imgur.com/USpNvYY.gif

So, it appears that the problem only occurs when I specify a particular _start_ frame. Trimming from the end works as expected.

I encountered the same problem with Gifsicle, so I don't think this is a bug with ImageMagick. I just can't figure out what would cause trimming to only break at the start but not at the end.

**Comment by fmw42** [comment]
FYI on ImageMagick 7, use magick, not magick convert.

**Reply by 3ternal** [reply]
Thanks, that's good to know. Doesn't solve the problem, but I'll update my bat script accordingly.

**Comment by dlemstra** [comment]
This is probably happening because the image is only doing partial updates in the frame. Your first frame contains all the information and this is missing from other frames. Examples of how that is done can be found here: https://legacy.imagemagick.org/Usage/anim_basics/. This can be resolved with doing a `-coalesce` but then you will first need to read all the frames and then delete some of them.

**Comment by snibgo** [comment, marked as answer]
As dlemstra says.

Your input GIFs are compresed, and you don't decompress them. That is why starting from some frame afte the first doesn't work.

For example, one of your inputs is:
```
f:\web\im>%IMG7%magick bPLrPYV.gif info:
bPLrPYV.gif[0] GIF 800x450 800x450+0+0 8-bit sRGB 256c 0.000u 0:00.005
bPLrPYV.gif[1] GIF 800x449 800x450+0+1 8-bit sRGB 256c 0.000u 0:00.019
bPLrPYV.gif[2] GIF 800x449 800x450+0+1 8-bit sRGB 256c 0.000u 0:00.032
bPLrPYV.gif[3] GIF 800x448 800x450+0+2 8-bit sRGB 256c 0.000u 0:00.044
bPLrPYV.gif[4] GIF 800x450 800x450+0+0 8-bit sRGB 256c 0.000u 0:00.059
bPLrPYV.gif[5] GIF 800x450 800x450+0+0 8-bit sRGB 256c 0.000u 0:00.078
bPLrPYV.gif[6] GIF 800x443 800x450+0+7 8-bit sRGB 128c 0.000u 0:00.094
bPLrPYV.gif[7] GIF 800x441 800x450+0+9 8-bit sRGB 128c 0.000u 0:00.107
bPLrPYV.gif[8] GIF 800x443 800x450+0+7 8-bit sRGB 256c 0.000u 0:00.120
bPLrPYV.gif[9] GIF 800x437 800x450+0+13 8-bit sRGB 256c 0.000u 0:00.133
bPLrPYV.gif[10] GIF 800x448 800x450+0+2 8-bit sRGB 256c 0.000u 0:00.146
bPLrPYV.gif[11] GIF 800x444 800x450+0+6 8-bit sRGB 256c 0.000u 0:00.158
bPLrPYV.gif[12] GIF 800x441 800x450+0+9 8-bit sRGB 256c 0.016u 0:00.174
bPLrPYV.gif[13] GIF 800x450 800x450+0+0 8-bit sRGB 128c 0.016u 0:00.194
bPLrPYV.gif[14] GIF 800x447 800x450+0+3 8-bit sRGB 256c 0.016u 0:00.214
bPLrPYV.gif[15] GIF 800x448 800x450+0+2 8-bit sRGB 128c 0.016u 0:00.230
bPLrPYV.gif[16] GIF 800x449 800x450+0+1 8-bit sRGB 256c 0.016u 0:00.242
bPLrPYV.gif[17] GIF 800x448 800x450+0+2 8-bit sRGB 128c 0.016u 0:00.254
bPLrPYV.gif[18] GIF 800x449 800x450+0+1 8-bit sRGB 256c 0.016u 0:00.268
bPLrPYV.gif[19] GIF 800x445 800x450+0+5 8-bit sRGB 128c 0.016u 0:00.280
bPLrPYV.gif[20] GIF 800x450 800x450+0+0 8-bit sRGB 128c 808128B 0.016u 0:00.293
```
The frames are all on a 800x450 canvas. But most images are smaller than that, due to the compression storing just changes between frames.

I suggest you first read all the frames, then "-layers coalesce", then delete the frames you don't want. (Or clone the frames you do want and delete all the others.)

**Reply by 3ternal** [reply]
Thanks for going into detail! This worked for me.

My command is now: `magick input.gif -layers coalesce -delete 0-25 output.gif`
